### PR TITLE
SC-1380

### PR DESCRIPTION
**Pull request by anmol2302**
<b>Ticket Ref:</b> SC-1380
1: Added /v1/user/migrate API to migrate user
2: userId and extUserId is mandatory also provided auth token should match with userId in request

### CODE REVIEW

File: service/app/controllers/tenantmigration/TenantMigrationController.java [hunk 1]
```diff
@@ -45,4 +46,20 @@
                 userId,
                 JsonKey.USER_ID, false);
     }
+
+    public Promise<Result> userSelfMigrate() {
```
Comment (manzarul) [inline service/app/controllers/tenantmigration/TenantMigrationController.java]: userSelfMigrate to shadowUserMigrate

File: service/app/controllers/usermanagement/validator/UserSelfMigrationReqValidator.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import org.sunbird.common.request.BaseRequestValidator;
+import org.sunbird.common.request.Request;
+import org.sunbird.common.responsecode.ResponseCode;
+
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class UserSelfMigrationReqValidator extends BaseRequestValidator {
```
Comment (manzarul) [inline service/app/controllers/usermanagement/validator/UserSelfMigrationReqValidator.java]: UserSelfMigrationReqValidator to ShadowUserMigrateReqValidator

File: service/app/controllers/tenantmigration/TenantMigrationController.java [hunk 3]
```diff
@@ -45,4 +46,20 @@
...
+        String tokenUserId=ctx().flash().get(JsonKey.USER_ID);
```
Comment (manzarul) [inline service/app/controllers/tenantmigration/TenantMigrationController.java]: tokenUserId to callerId

Comment (anmol2302) [reply service/app/controllers/tenantmigration/TenantMigrationController.java]: @manzarul done

File: service/app/controllers/usermanagement/validator/UserSelfMigrationReqValidator.java (hunk 2 above)
Comment (anmol2302) [reply service/app/controllers/usermanagement/validator/UserSelfMigrationReqValidator.java]: @manzarul done

File: service/app/controllers/tenantmigration/TenantMigrationController.java (hunk 1 above)
Comment (anmol2302) [reply service/app/controllers/tenantmigration/TenantMigrationController.java]: @manzarul done

**Pull request merged**